UploadDaemon_Test/UploadTaskTest.cs

```diff
@@ -247,9 +264,24 @@ public Task<bool> UploadLineCoverageAsync(string originalTraceFilePath, string l
 
     private class MockLineCoverageSynthesizer : ILineCoverageSynthesizer
     {
+        private readonly bool shouldProduceCoverage;
+
+        public MockLineCoverageSynthesizer() : this(true)

```

Comment:
You could use an optional parameter for this (`bool shouldProduceCoverage = true`).